falconjs/packages/falcon/routes/frontendOnlyRoutes.js

```diff
@@ -0,0 +1,99 @@
+const pathToRegexp = require('path-to-regexp');
+const debug = require('debug')('falcon:routing/frontendOnlyRoutes');
+const { parse } = require('url');
+
+/**
+ * @file
+ * This file defines middlewares and helpers for app only route.
+ */
+
+/**
+ * Finds internal route in the pages available at the frontend app only.
+ * @param URL
+ *   Full URL string. I.e. /content/one/two?page=1.
+ * @param routes
+ *   List of routes.
+ * @returns {*}
+ *   Matched route & route query with dynamic values from the URL.
+ */
+const matchAppOnlyRoute = (URL, routes) => {
+  // Get object with different parts of the URL.
+  const parsedUrl = parse(URL, true);
+
+  // By default pass into the request the same query details as the
+  // browser URL has.
+  const hrefQuery = parsedUrl.query;
+
+  // Finds matching route using regexp.
+  const machingRoute = routes.find((route) => {
+    const keys = [];
+    const regex = pathToRegexp(route.path, keys);
+    let values = regex.exec(parsedUrl.pathname);
+    if (values) {
+      // Remove the first array value as it always contains the full url string.
+      values = values.slice(1);
+
+      // Adds dynamic parts of the url to the query object.
+      keys.forEach((key, index) => {
+        hrefQuery[key.name] = decodeURIComponent(values[index]);
+      });
+
+      return true;
+    }
+    return false;
+  });
+
+  // Add a little flag that tells _app.js to not make a request to the backend
+  // searching for the matching entity.
+  hrefQuery.isAppOnlyRoute = true;
+
+  if (machingRoute) {
+    return { href: machingRoute.route, query: hrefQuery };
+  }
+
+  return null;
+};
+
+/**
+ * Converts object of URL query values into query string
+ * valid for the browser.
+ */
+const toQuerystring = obj => Object.keys(obj)
+  .filter(key => obj[key] !== null && obj[key] !== undefined)
+  .map((key) => {
+    let value = obj[key];
+
+    if (Array.isArray(value)) {
+      value = value.join('/');
+    }
+    return [
+      encodeURIComponent(key),
+      encodeURIComponent(value),
+    ].join('=');
+  }).join('&');
+
+
+function frontendOnlyRoutes(nextApp, routes) {
```

Comment:
We seem to have a mix of named functions with ES6 function syntax (e.g. line 61) alongside these anonymous declarations in several files. Wouldn't it better if we made all function declarations consistent with ES6 syntax in the `const` style?

Reply:
An example with `async`: 
```
const funcName = async (param) => {};
```